### PR TITLE
type_5_object_ops

### PR DESCRIPTION
**Pull request by VikasKumarC**
Signed-off-by: Vikas Kumar Choudhary <vikas.kumar@seagate.com>

## Problem Statement

-   Problem statement

## Design

-   For Bug, Describe the fix here.
-   For new test addition post the link for design

## Coding

   Checklist for Author

-   [ ]  Coding conventions are followed and code is formatted (e.g. using pycharm in-built formatter)

## Testing

  Checklist for Author

-   [ ]  New/Affected tests are executed
-   [ ]  Attach test execution logs

## Review 

  Checklist for Author

-   [ ]  JIRA number/GitHub Issue added to PR
-   [ ]  PR is self reviewed
-   [ ]  Jira and state/status is updated and JIRA is updated with PR link
-   [ ]  Check if the description is clear and explained

## Documentation

  Checklist for Author

-   [ ]  Changes done to ReadMe

### CODE REVIEW

File: docs/environment_setup/Dev_Environment_Setup [hunk 1]
```diff
@@ -1,9 +1,11 @@
 #Specific to Cent OS and RHEL
 yum update -y
-yum install -y gcc openssl-devel bzip2-devel libffi-devel zlib-devel wget make sqlite-devel
+yum install -y wget make
+yum  install -y *sqlite* --skip-broken
+yum install -y *lzma*
+yum install -y gcc openssl-devel bzip2-devel libffi-devel zlib-devel sqlite-devel xz-devel
```
Comment (DhobaleVishal) [inline docs/environment_setup/Dev_Environment_Setup]: Add comments why this packages are required.

File: docs/environment_setup/Dev_Environment_Setup [hunk 2]
```diff
@@ -16,6 +18,15 @@ make install
 Create a soft link to point installation
 ln -s /usr/local/bin/python3.7 python3.7
 
+# following aliases may be added in /root/.bashrc
+
```
Comment (DhobaleVishal) [inline docs/environment_setup/Dev_Environment_Setup:24]: remove empty line.

**Review by DhobaleVishal — CHANGES_REQUESTED** [review]

File: docs/environment_setup/Dev_Environment_Setup (hunk 1 above)
Comment (VikasKumarC) [reply docs/environment_setup/Dev_Environment_Setup]: Done

File: docs/environment_setup/Dev_Environment_Setup (hunk 2 above)
Comment (VikasKumarC) [reply docs/environment_setup/Dev_Environment_Setup:24]: removed

File: src/libs/s3api/s3io_utils.py [hunk 3]
```diff
@@ -255,3 +261,22 @@ def get_random_sleep_time(delay) -> int:
         else:
             sleep_time = delay
         return sleep_time
+
+    async def overwrite_distribution_data(self, distribution, object_size) -> None:
+        """overwrite 10% of total objects in a given bucket"""
```
Comment (DhobaleVishal) [inline src/libs/s3api/s3io_utils.py]: Is overwrite fixed to 10%?

File: scripts/s3/s3api/bucket_objects_operations.py [hunk 4]
```diff
@@ -60,23 +60,31 @@ async def execute_bucket_object_workload(self):
...
-                time.sleep(sleep_time)
-                self.starts_sessions(self.read_data, distribution)
-                self.starts_sessions(self.delete_distribution_data, distribution)
-                self.starts_sessions(self.write_distribution_data, distribution, object_size)
+                if delay:
+                    sleep_time = self.get_random_sleep_time(delay)
+                    self.log.info("sleep for %s hrs", sleep_time / (60 ** 2))
+                    time.sleep(sleep_time)
+                if overwrite_percentage_per_bucket:
+                    self.starts_sessions(self.overwrite_data, overwrite_percentage_per_bucket )
```
Comment (DhobaleVishal) [inline scripts/s3/s3api/bucket_objects_operations.py]: correct the method name to 'overwrite_distribution_data'

Comment (DhobaleVishal) [inline scripts/s3/s3api/bucket_objects_operations.py]: 'overwrite_distribution_data' method accepts 'distribution, object_size' parameters, pls correct them.

File: scripts/s3/s3api/bucket_objects_operations.py [hunk 5]
```diff
@@ -60,23 +60,31 @@ async def execute_bucket_object_workload(self):
             number_of_objects = self.kwargs.get("number_of_objects")
             delete_percentage_per_bucket = self.kwargs.get("delete_percentage_per_bucket")
             put_percentage_per_bucket = self.kwargs.get("put_percentage_per_bucket")
+            overwrite_percentage_per_bucket = self.kwargs.get("overwrite_percentage_per_bucket")
```
Comment (DhobaleVishal) [inline scripts/s3/s3api/bucket_objects_operations.py:63]: I can see overwrite distribution is not calculated. For same refer line no 70 and make it conditional.

**Review by DhobaleVishal — CHANGES_REQUESTED** [review]
@VikasKumarC Please check inline comments.

File: src/libs/s3api/s3io_utils.py (hunk 3 above)
Comment (VikasKumarC) [reply src/libs/s3api/s3io_utils.py]: Done

File: scripts/s3/s3api/bucket_objects_operations.py (hunk 4 above)
Comment (VikasKumarC) [reply scripts/s3/s3api/bucket_objects_operations.py]: Done

File: scripts/s3/s3api/bucket_objects_operations.py (hunk 5 above)
Comment (VikasKumarC) [reply scripts/s3/s3api/bucket_objects_operations.py:63]: Done

**Review by DhobaleVishal — APPROVED** [review]
LGTM

**Pull request merged**